scene/gui/control.cpp

```diff
@@ -128,10 +128,7 @@ bool Control::_edit_use_rotation() const {
 }
 
 void Control::_edit_set_pivot(const Point2 &p_pivot) {
-	Vector2 delta_pivot = p_pivot - get_pivot_offset();
-	Vector2 move = Vector2((cos(data.rotation) - 1.0) * delta_pivot.x - sin(data.rotation) * delta_pivot.y, sin(data.rotation) * delta_pivot.x + (cos(data.rotation) - 1.0) * delta_pivot.y);
-	set_position(get_position() + move);
-	set_pivot_offset(p_pivot);
+	set_pivot_offset(get_transform().xform(p_pivot));
```

Comment:
No this should definitely stay like it was. It is normal that in the _edit_set_pivot function that the node's position is changed. With your solution, if you set the pivot via the editor (with the V key), the node moves on the screen.
It's something that is less UX friendly than before.

Reply:
I was thinking about this a lot.
To some extend I agree. It is more obvious in the first place. changing pivot leaves the node at the same position and the rotation pivot is different after the operation.

But I feel like changing the pivot really only should change the pivot, to reflect what changing pivot does when doing it manually in code.
I guess most ppl need a little bit of time to wrap their heads around the pivot/position concept and why the objects seems to move when the pivot is changed and the rotation != 0.

So it is helpful when the editor already represents that behaviour. Playing around with it makes it really easy to figure out how the pivot behaves and you learn to deal with it. when changing the pivot in game you already know what you do and are not completely confused.

It also can lead to
TLDR;
the editor should represent api pivot operations.

Reply:
No, I don't think it should.
The canvas item editor is a visual tool, it is meant to be easy to use, with expected visible results. If the node moves when you move the pivot it definitely complexifies the workflow, and makes things hard to understand.

To sum up my point: if you use visual tools, the result should be as obvious as possible, even if it screws up the position or other values. This should obviously not happen when you use the inspector, where changing the pivot should not change the position.

Reply:
Okay.
So the '_edit' function. Is doing it's special thing with fixing the position after pivot change.
But changing only pivot, only changes pivot!
Thats sounds reasonable!!

Reply:
Exactly :)